### PR TITLE
Feature: add campaign goal types for recurring donations

### PR DESCRIPTION
**Pull request by glaubersilva**
Resolves [GIVE-1265]

## Description

This PR add three new goal types related to recurring donations.

## Affects

New campaign creating and editing (settings tab)

## Visuals

image

## Testing Instructions

Create and edit new campaigns using 3 new options for recurring donations.

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [ ] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [ ] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

[GIVE-1265]: https://stellarwp.atlassian.net/browse/GIVE-1265?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Campaigns/resources/admin/components/CampaignDetailsPage/Tabs/Settings.tsx [hunk 1]
```diff
@@ -111,8 +123,11 @@ export default () => {
 
                         <select {...register('goalType')}>
                             <option value="amount">{__('Amount raised', 'give')}</option>
-                            <option value="donation">{__('Number of Donations', 'give')}</option>
-                            <option value="donors">{__('Number of Donors', 'give')}</option>
+                            <option value="donations">{__('Number of donations', 'give')}</option>
+                            <option value="donors">{__('Number of donors', 'give')}</option>
+                            <option value="amountFromSubscriptions">{__('Recurring amount raised', 'give')}</option>
```
Comment (alaca) [inline src/Campaigns/resources/admin/components/CampaignDetailsPage/Tabs/Settings.tsx:128]: These should be displayed only if the recurring add-on is installed.

File: src/Campaigns/resources/admin/components/CampaignFormModal/index.tsx [hunk 2]
```diff
@@ -274,25 +316,55 @@ export default function CampaignFormModal({isOpen, handleClose, apiSettings, tit
...
-                                    label={__('Number of Donors', 'give')}
+                                    label={__('Number of donors', 'give')}
                                     description={__(
                                         'Your goal progress is measured by the number of donors. eg. 10 of 50 donors have given.',
                                         'give'
                                     )}
                                     selected={selectedGoalType === 'donors'}
                                     register={register}
                                 />
+                                <GoalTypeOption
```
Comment (alaca) [inline src/Campaigns/resources/admin/components/CampaignFormModal/index.tsx:338]: Same here. I'd expect these to be visible only if the recurring add-on is installed.

**Review by alaca — COMMENTED** [review]
Hey, man! Good job. I have only one thing that should be addressed - showing recurring fields only if the recurring add-on is installed. Other than that, this is good to go.

File: src/Campaigns/resources/admin/components/CampaignDetailsPage/Tabs/Settings.tsx (hunk 1 above)
Comment (glaubersilva) [reply src/Campaigns/resources/admin/components/CampaignDetailsPage/Tabs/Settings.tsx]: Resolved in 373be0c05a118d99af9801c02e4d66a97e196425

File: src/Campaigns/resources/admin/components/CampaignFormModal/index.tsx (hunk 2 above)
Comment (glaubersilva) [reply src/Campaigns/resources/admin/components/CampaignFormModal/index.tsx]: Resolved in 373be0c05a118d99af9801c02e4d66a97e196425

**Comment by glaubersilva** [comment]
@alaca Nice catch about the recurring add-on, thanks! This is ready for re-review.

File: src/Campaigns/resources/admin/components/CampaignDetailsPage/Tabs/Settings.tsx [hunk 3]
```diff
@@ -111,8 +123,19 @@ export default () => {
 
                         <select {...register('goalType')}>
                             <option value="amount">{__('Amount raised', 'give')}</option>
-                            <option value="donation">{__('Number of Donations', 'give')}</option>
-                            <option value="donors">{__('Number of Donors', 'give')}</option>
+                            <option value="donations">{__('Number of donations', 'give')}</option>
+                            <option value="donors">{__('Number of donors', 'give')}</option>
+                            {window.GiveCampaignDetails.isRecurringEnabled && (
```
Comment (alaca) [inline src/Campaigns/resources/admin/components/CampaignDetailsPage/Tabs/Settings.tsx:128]: All option components can be inside one condition, just wrap them with `fragment`. 

Example:
```js
{condition && (
  <>
    <option />
    <option />
    <option />
  </>
)}
```

File: src/Campaigns/resources/admin/components/CampaignFormModal/index.tsx [hunk 4]
```diff
@@ -274,25 +319,61 @@ export default function CampaignFormModal({isOpen, handleClose, apiSettings, tit
...
-                                    label={__('Number of Donors', 'give')}
+                                    label={__('Number of donors', 'give')}
                                     description={__(
                                         'Your goal progress is measured by the number of donors. eg. 10 of 50 donors have given.',
                                         'give'
                                     )}
                                     selected={selectedGoalType === 'donors'}
                                     register={register}
                                 />
+                                {isRecurringEnabled && (
```
Comment (alaca) [inline src/Campaigns/resources/admin/components/CampaignFormModal/index.tsx:341]: Same here. There is no need to have each option in a separate condition.

**Review by alaca — COMMENTED** [review]
Hey man, much better.
Just put everything in one condition and this is good to go.

File: src/Campaigns/resources/admin/components/CampaignDetailsPage/Tabs/Settings.tsx (hunk 3 above)
Comment (glaubersilva) [reply src/Campaigns/resources/admin/components/CampaignDetailsPage/Tabs/Settings.tsx:128]: Resolved in 616b9cf7a1c68f33ee593b839d4d98de5230918c

File: src/Campaigns/resources/admin/components/CampaignFormModal/index.tsx (hunk 4 above)
Comment (glaubersilva) [reply src/Campaigns/resources/admin/components/CampaignFormModal/index.tsx:341]: Resolved in 616b9cf7a1c68f33ee593b839d4d98de5230918c

**Comment by glaubersilva** [comment]
@alaca Thanks, man! Everything is in one condition now.

**Review by alaca — APPROVED** [review]
Yeah, it's much cleaner now. Thanks, man. Good job 👍

**Pull request merged**